web/src/lib/components/shared-components/change-date.svelte

```diff
@@ -24,10 +24,19 @@
     value: string;
   };
 
-  const timezones: ZoneOption[] = Intl.supportedValuesOf('timeZone').map((zone: string) => ({
-    label: zone + ` (${DateTime.local({ zone }).toFormat('ZZ')})`,
-    value: 'UTC' + DateTime.local({ zone }).toFormat('ZZ'),
-  }));
+  const timezones: ZoneOption[] = Intl.supportedValuesOf('timeZone')
+    .sort((zoneA, zoneB) => {
+      let numericallyCorrect = DateTime.local({ zone: zoneA }).offset - DateTime.local({ zone: zoneB }).offset;
```

Comment:
You could do something like this so you're only calling `DateTime.local()` once:
```typescript
  const timezones: ZoneOption[] = Intl.supportedValuesOf('timeZone')
  	.map((zone) => DateTime.local(zone))
    .sort((zoneA, zoneB) => {
      let numericallyCorrect = zoneA.offset - zoneB.offset;
      ...etc
```

Reply:
But you will have to pass the zone string for the creation of the list items in the last .map
so it will be something more like this
```ts
  const timezones: ZoneOption[] = Intl.supportedValuesOf('timeZone')
    .map((zone) => ({
      datatime: DateTime.local({ zone }),
      string: zone,
    }))
    .sort((zoneA, zoneB) => {
      let numericallyCorrect = zoneA.datatime.offset - zoneB.datatime.offset;
      if (numericallyCorrect != 0) {
        return numericallyCorrect;
      }
      const alphabeticallyCorrect = zoneA.string.localeCompare(zoneB.string, undefined, { sensitivity: 'base' });
      return alphabeticallyCorrect;
    })
    .map((zone) => ({
      label: zone.string + ` (${DateTime.local({ zone: zone.string }).toFormat('ZZ')})`,
      value: 'UTC' + DateTime.local({ zone: zone.string }).toFormat('ZZ'),
    }));
```

Reply:
`DateTime.zoneName` can be used:

```ts
const timezones: ZoneOption[] = Intl.supportedValuesOf('timeZone')
  .map((zone) => DateTime.local({ zone }))
  .sort((a, b) => a.offset - b.offset || a.zoneName.localeCompare(b.zoneName, undefined, { sensitivity: 'base' }))
  .map((date) => {
    const offset = date.toFormat('ZZ');
    return {
      label: `${date.zoneName} (${offset})`,
      value: 'UTC' + offset,
    };
  });
```